hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/builder/tasks/SearchTask.java

```diff
@@ -593,6 +593,7 @@ private void doSearch() {
 
 			if (next == -1) {
 				sb.setMaxResultsToFetch(null);
+				sb.setShouldDeduplicateInDB(true);
```

Comment:
I guess this is the most important line in the whole thing, lol... Recommend flagging with a comment
```suggestion
				// If we're past the last prefetch threshold then we're fetching potentially
				// unlimited amounts of data. We'll move responsibility for deduplication
				// to the database in this case so that we don't run the rist of blowing out
				// the memory in the app server
				sb.setShouldDeduplicateInDB(true);
```

Reply:
added